src/content/docs/explanations/language/expressions.mdx

```diff
@@ -0,0 +1,1175 @@
+
+```tql
+where timestamp > now() - 1d and severity == "critical"
+where port == 22 or port == 3389
+```
+
+These operators use short-circuit evaluation.
+
+#### Membership Testing (`in`)
+
+The `in` operator tests containment across different types:
+
+| Expression            | Checks if...                            |
+| :-------------------- | :-------------------------------------- |
+| `value in list`       | List contains the value                 |
+| `substring in string` | String contains the substring           |
+| `ip in subnet`        | IP address is within the subnet range   |
+| `subnet in subnet`    | First subnet is contained in the second |
+
+```tql
+from {
+  ip: 10.0.0.5,
+  port: 443,
+  message: "connection error",
+  in_private: ip in 10.0.0.0/8,
+  is_https: port in [443, 8443],
+  has_error: "error" in message,
+}
+```
+
+```tql
+{
+  ip: 10.0.0.5,
+  port: 443,
+  message: "connection error",
+  in_private: true,
+  is_https: true,
+  has_error: true,
+}
+```
+
+To negate membership tests, use `not in` or `not (value in container)`.
+
+### Operator Precedence
+
+Operations follow standard precedence rules:
+
+| Precedence  | Operators                                | Associativity |
+| ----------- | ---------------------------------------- | ------------- |
+| 1 (highest) | Method call, field access, `[]` indexing | -             |
+| 2           | Unary `+`, `-`                           | -             |
+| 3           | `*`, `/`                                 | Left          |
+| 4           | Binary `+`, `-`                          | Left          |
+| 5           | `==`, `!=`, `<`, `<=`, `>`, `>=`, `in`   | Left          |
+| 6           | `not`                                    | -             |
+| 7           | `and`                                    | Left          |
+| 8 (lowest)  | `or`                                     | Left          |
+
+Expressions like `1 - 2 * 3 + 4` follow these precedence and associativity rules.
+The expression evaluates as `(1 - (2 * 3)) + 4`. Example: `1 + 2 * 3` evaluates as `1 + (2 * 3)` = 7
+
+## Conditional Expressions
+
+### Python-style Conditionals
+
+TQL uses Python-style conditional expressions, i.e., `x if condition else y`
+where `x`, `y`, and `condition` are expressions.
+
+Use conditionals in assignments and format strings:
+
+```tql
+from {
+  response_code: 200,
+  success: true,
+  status: "OK" if response_code == 200 else "ERROR",
+  message: f"Status: {'✓' if success else '✗'}",
+}
+```
+
+Chaining is allowed but discouraged for readability:
+
+```tql
+from {
+  severity: "high",
+  priority: 1 if severity == "critical" else 2 if severity == "high" else 3,
+}
+```
+
+### Standalone `if`
+
+`if` acts as a guard, returning `null` when false:
+
+```tql
+from {
+  performance: "good",
+  should_compute: false,
+  bonus: 1000 if performance == "excellent",  // null otherwise
+  result: now() if should_compute,            // null since should_compute is false
+```
```

Comment:
```suggestion
```tql
from {
  performance: "good",
  should_compute: false,
  bonus: 1000 if performance == "excellent",  // null otherwise
  result: now() if should_compute,            // null since should_compute is false
}
```
```